eng/packaging.targets

```diff
@@ -11,6 +11,7 @@
     <!-- Don't include target platform specific dependencies, since we use the target platform to represent RIDs instead -->
     <SuppressDependenciesWhenPacking Condition="'$(TargetPlatformIdentifier)' != ''">true</SuppressDependenciesWhenPacking>
     <PackageDesignerMarkerFile>$(MSBuildThisFileDirectory)useSharedDesignerContext.txt</PackageDesignerMarkerFile>
+    <PackageReadmeFile Condition="'$(PackageReadmeFile)' == '' and Exists('PACKAGE.md')">PACKAGE.md</PackageReadmeFile>
```

Comment:
Maybe the file name could be `PACKAGE_README.md`? `PACKAGE.md` doesn't really describe exactly what the file is.

Reply:
It's more about readability/understandability of the files in the repo.

Reply:
Other options include:

1. `PackageReadme.md`
2. The file under a `docs` folder, and still name it `README.md`. e.g. `src/docs/README.md`
3. Since the files are split already - one is in the root of the library, and one is under `src` - they could both be named `README.md`.

Reply:
The PACKAGE.md choice was based on this suggestion: https://github.com/dotnet/runtime/issues/59630#issuecomment-1324517038 I don't think this name needs to be long as there's not a lot of .md files in repo to disambiguate between them.  So personally i'd prefer either README.md or PACKAGE.md.

Reply:
Being able to distinguish between a package documentation file and a repo specific contributing guidance seems valuable to me. We unintentionally started shipping the contributing guidance files to customers and this PR fixes that: https://github.com/dotnet/runtime/issues/59630#issuecomment-1323411291.